### PR TITLE
Add group() aggregator

### LINKED ISSUE #7400: `any()` aggregation operator
## `any()` aggregation operator

It would be useful if we had an aggregation operator that explicitly states "i don't care which value you pick, just pick one". This could have a marginal performance advantage over choosing an arbitrary aggregation, but the real benefit is semantic. when you put, for example, `min()` in a query, the reader assumes that this particular aggregation was chosen for a reason, but in some cases it doesn't matter, like when multiple identical or non-overlapping series are being aggregated. picking an arbitrary aggregation adds meaningless specificity and can make it harder to understand complex queries, while something like `any()` would remove any false meaning from the expression.

### PR DESCRIPTION
**Pull request by roidelapluie**
Fix #7400 
cc @alice-sawatzky @beorn7 

Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

### CODE REVIEW

File: promql/engine.go [hunk 1]
```diff
@@ -2004,6 +2007,8 @@ func (ev *evaluator) aggregation(op parser.ItemType, grouping []string, without
 			group.groupCount++
 			group.mean += (s.V - group.mean) / float64(group.groupCount)
 
+		case parser.GROUP:
```
Comment (brian-brazil) [inline promql/engine.go:2010]: I'd add a comment to make it clear this purposefully does nothing.

**Comment by beorn7** [comment]
I like it. Thank you very much.

Could you also add documentation?

**Comment by roidelapluie** [comment]
> I like it. Thank you very much.
> 
> Could you also update the documentation?

It is updated, what else would you add ?

File: docs/querying/operators.md [hunk 2]
```diff
@@ -187,6 +187,7 @@ vector of fewer elements with aggregated values:
 * `min` (select minimum over dimensions)
 * `max` (select maximum over dimensions)
 * `avg` (calculate the average over dimensions)
+* `group` (returns 1 over dimensions)
```
Comment (roidelapluie) [inline docs/querying/operators.md:190]: @beorn7

Comment (beorn7) [reply docs/querying/operators.md:190]: Sorry, missed that. Perhaps it's a bit dense? Even I needed to think a bit about what "returns 1 over dimensions" means.

Perhaps "(all values in the resulting vector are 1)"?

Comment (roidelapluie) [reply docs/querying/operators.md]: Adapted

**Pull request merged**